### PR TITLE
#10: make int tests independent

### PR DESCRIPTION
**Pull request by piotrlg9**
MongoDB instance does not have to be running for tests after this change.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -12,6 +12,8 @@
         <maven.compiler.source>17</maven.compiler.source>
         <maven.compiler.target>17</maven.compiler.target>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <maven-surefire-plugin.version>2.22.0</maven-surefire-plugin.version>
```
Comment (piotr-bugara-gravity9) [inline pom.xml:15]: Why not the latest version - `3.2.3`?

File: src/test/java/com/gravity9/mongocdc/ChangeStreamTest.java [hunk 2]
```diff
@@ -5,33 +5,43 @@
...
+	private static final Logger log = LoggerFactory.getLogger(ChangeStreamTest.class);
+
+	@BeforeEach
+	public void setup() {
+		super.setup();
+		mongoConfig = MongoConfig.builder()
+				.connectionUri(getConnectionUri())
+				.databaseName(getDatabaseName())
+				.collectionName(getTestCollectionName())
+				.workerConfigCollectionName("changeStreamWorkerConfig")
```
Comment (piotr-bugara-gravity9) [inline src/test/java/com/gravity9/mongocdc/ChangeStreamTest.java:33]: Use `getWorkerConfigCollectionName()` and `getClusterConfigCollectionName()` methods

File: src/test/java/com/gravity9/mongocdc/ChangeStreamTest.java [hunk 3]
```diff
@@ -45,14 +55,15 @@ void givenProperConfiguration_shouldStartManagerAndRegisterListener() throws Exc
...
+
 		Document testDoc = new Document("testValue", 123);
-		collection.insertOne(testDoc);
+		InsertOneResult insertOneResult = collection.insertOne(testDoc);
+		log.info("Inserted document: {}", insertOneResult.getInsertedId());
 
-		// Wait for CDC event to arrive
-		Thread.sleep(1000);
+		List<ChangeStreamDocument<Document>> events;
+		events = waitForEvents(listener);
```
Comment (piotr-bugara-gravity9) [inline src/test/java/com/gravity9/mongocdc/ChangeStreamTest.java:65]: change it to `List<ChangeStreamDocument<Document>> events = waitForEvents(listener);`

File: src/test/java/com/gravity9/mongocdc/ConfigManagerTest.java [hunk 4]
```diff
@@ -1,27 +1,25 @@
...
-	void cleanUp() {
-		TestMongoUtils.cleanUp();
+	@BeforeEach
+	public void setup() {
+		super.setup();
+		mongoConfigBuilder = new MongoConfig.MongoConfigBuilder()
+				.connectionUri(getConnectionUri())
+				.databaseName(getDatabaseName())
+				.collectionName(getTestCollectionName())
+				.workerConfigCollectionName("changeStreamWorkerConfig")
```
Comment (piotr-bugara-gravity9) [inline src/test/java/com/gravity9/mongocdc/ConfigManagerTest.java:21]: the same as in previous comment - use methods from AbstractMongoDbBase

**Review by piotr-bugara-gravity9 — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (piotrlg9) [reply pom.xml]: All plugins updated

File: src/test/java/com/gravity9/mongocdc/ChangeStreamTest.java (hunk 2 above)
Comment (piotrlg9) [reply src/test/java/com/gravity9/mongocdc/ChangeStreamTest.java]: Fixed

File: src/test/java/com/gravity9/mongocdc/ChangeStreamTest.java (hunk 3 above)
Comment (piotrlg9) [reply src/test/java/com/gravity9/mongocdc/ChangeStreamTest.java]: Fixed

File: src/test/java/com/gravity9/mongocdc/ConfigManagerTest.java (hunk 4 above)
Comment (piotrlg9) [reply src/test/java/com/gravity9/mongocdc/ConfigManagerTest.java]: Fixed

**Review by piotr-bugara-gravity9 — APPROVED** [review]

**Pull request merged**